Fix crash when NPCNames is empty

Maeyanie opened the pull request:
Fixes this crash I encountered, and apparently so did someone in your comments:
```
Assigning item types and prepping things ...
Patching for Potion of Blood ...
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at Hephaestus.Program.RunPatch(IPatcherState`2 state) in F:\tmp\Synthesis\rvmcfdzj.l05\Git\fbx413i3.pfn\Runner\Hephaestus\Program.cs:line 391
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass15_0`2.<<AddPatch>b__0>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 103
--- End of stack trace from previous location ---
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args, Nullable`1 exportKey, IFileSystem fileSystem) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 607
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass43_0.<<Run>b__0>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 472
--- End of stack trace from previous location ---
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.HandleOnShutdown(Func`1 a) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 833
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args, IFileSystem fileSystem) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 460
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass42_0.<<InternalRun>b__1>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\Pipeline\SynthesisPipeline.cs:line 432
```

Maeyanie committed:
Fix crash when NPCNames is empty

I'm not sure exactly how this happens, but apparently it can.

CritLoren commented:
oh curious, the code is already meant to account for this via the if/else schema where it's building the NPCNames list by having some generic NPC names to begin with added to it, Only reason for it to not exist is if the item seemingly has no keywords at all I think, does the potion of blood have no keywords?

Maeyanie commented:
That does appear to be the case, yes:

![image](https://github.com/user-attachments/assets/3d752a01-fb71-4c85-81e3-ce70e0fc6d08)

CritLoren committed:
Add else statement instead

Instead of trying to catch the error at the end, extend the if/else function to correctly use the entire list of available names as intended

CritLoren commented:
give that a try, should correctly assign a name. I kept the if else you added though, as you never have too many backup plans.

Maeyanie commented:
That seems to have worked:

![image](https://github.com/user-attachments/assets/ca043d92-0a5d-45e7-a44d-395b5600a82a)

Completely agree though, it never hurts to have an extra fallback, just in case.

CritLoren commented:
Lovely, thank you for the contribution :D
